File: boj/6603_로또.py
import sys

sys.stdin = open("./input.txt", 'r')

input_data = sys.stdin.readlines()


def combination(index, level):
    global arr, result, k
    if level == 6:
        print(' '.join(map(str, arr)))
        return

    for i in range(index, k):
        arr.append(s[i])
        combination(i+1, level+1)
        arr.pop()


result = []

for array in input_data:
    arr = []
    data = list(map(int, array.split()))
    k = data[0]
    s = data[1:]

    if data[0] == 0:
        break

    combination(0, 0)
    print()

File: boj/6603_로또.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = require('fs').readFileSync(filePath).toString().trim().split('\n');
input.pop();
function solution(input) {
	const result = [];
	input.forEach((element) => {
		const [k, ...s] = element.split(' ').map(Number);
		console.log('🚀 ~ input.forEach ~ k:', k);

		const arr = [];

		const combination = (index, level) => {
			if (level === 6) {
				result.push(arr.join(' '));
				return;
			} else {
				for (let i = index; i < k; i += 1) {
					arr.push(s[i]);
					combination(i + 1, level + 1);
					arr.pop();
				}
			}
		};

		combination(0, 0);
		result.push('');
	});

	return result.join('\n');
}

console.log(solution(input));
